play-json/shared/src/test/scala/JsonValidSharedSpec.scala

```diff
@@ -3,68 +3,68 @@
-      JsNumber(5).validate[String] must equalTo(JsError(Seq(JsPath() -> Seq(JsonValidationError("error.expected.jsstring")))))
-      JsNumber(5.123).validate[Int] must equalTo(JsError(Seq(JsPath() -> Seq(JsonValidationError("error.expected.int")))))
-      JsNumber(300).validate[Byte] must equalTo(JsError(Seq(JsPath() -> Seq(JsonValidationError("error.expected.byte")))))
-      JsNumber(Long.MaxValue).validate[Int] must equalTo(JsError(Seq(JsPath() -> Seq(JsonValidationError("error.expected.int")))))
-      JsBoolean(false).validate[Double] must equalTo(JsError(Seq(JsPath() -> Seq(JsonValidationError("error.expected.jsnumber")))))
+      JsString("string").validate[Long] mustEqual (JsError(Seq(JsPath() -> Seq(JsonValidationError("error.expected.jsnumber")))))
+      JsNumber(5).validate[String] mustEqual (JsError(Seq(JsPath() -> Seq(JsonValidationError("error.expected.jsstring")))))
+      JsNumber(5.123).validate[Int] mustEqual (JsError(Seq(JsPath() -> Seq(JsonValidationError("error.expected.int")))))
+      JsNumber(300).validate[Byte] mustEqual (JsError(Seq(JsPath() -> Seq(JsonValidationError("error.expected.byte")))))
+      JsNumber(Long.MaxValue).validate[Int] mustEqual (JsError(Seq(JsPath() -> Seq(JsonValidationError("error.expected.int")))))
+      JsBoolean(false).validate[Double] mustEqual (JsError(Seq(JsPath() -> Seq(JsonValidationError("error.expected.jsnumber")))))
     }
 
     "validate simple numbered type conversion" in {
-      JsNumber(5).validate[Double] must equalTo(JsSuccess(5.0))
-      JsNumber(BigDecimal(5)).validate[Double] must equalTo(JsSuccess(5.0))
-      JsNumber(5.123).validate[BigDecimal] must equalTo(JsSuccess(BigDecimal(5.123)))
+      JsNumber(5).validate[Double] mustEqual (JsSuccess(5.0))
+      JsNumber(BigDecimal(5)).validate[Double] mustEqual (JsSuccess(5.0))
+      JsNumber(5.123).validate[BigDecimal] mustEqual (JsSuccess(BigDecimal(5.123)))
     }
 
     "return JsResult with correct values for isSuccess and isError" in {
-      JsString("s").validate[String].isSuccess must beTrue
-      JsString("s").validate[String].isError must beFalse
-      JsString("s").validate[Long].isSuccess must beFalse
-      JsString("s").validate[Long].isError must beTrue
+      JsString("s").validate[String].isSuccess mustEqual (true)
+      JsString("s").validate[String].isError mustEqual (false)
+      JsString("s").validate[Long].isSuccess mustEqual (false)
+      JsString("s").validate[Long].isError mustEqual (true)
     }
 
     "validate JsObject to Map" in {
-      Json.obj("key1" -> "value1", "key2" -> "value2").validate[Map[String, String]] must equalTo(JsSuccess(Map("key1" -> "value1", "key2" -> "value2")))
-      Json.obj("key1" -> 5, "key2" -> 3).validate[Map[String, Int]] must equalTo(JsSuccess(Map("key1" -> 5, "key2" -> 3)))
-      Json.obj("key1" -> 5.123, "key2" -> 3.543).validate[Map[String, Float]] must equalTo(JsSuccess(Map("key1" -> 5.123F, "key2" -> 3.543F)))
```

Comment:
This test with float values is moved to the JVM specific suite, as the behaviour of ScalaJS is different about that